docs/user-manual/graphics/shaders/wgsl-specifics.md

```diff
@@ -3,4 +3,273 @@ title: WGSL Specifics
+
+```wgsl
+attribute aUv0: vec2f;
+```
+
+Internally, a `VertexInput` struct is automatically created and populated with all the attributes. Attributes can be accessed from the structure passed to the main function, but also in the global scope.
+
+```wgsl
+attribute aUv0: vec2f;
+
+@vertex fn vertexMain(input: VertexInput) -> VertexOutput {
+
+    // access it using input passed to the main function
+    var myUv1 = input.aUv0;
+
+    // but also as a global variable (particularly useful inside other functions)
+    var myUv2 = aUv0;
+}
+```
+
+As part of the VertexInput structure, and also in the global scope, these built-in attributes are automatically available:
+
+```wgsl
+vertexIndex: @builtin(vertex_index)
+instanceIndex: @builtin(instance_index)
+```
+
+The attribute names must match the names specified in the `attributes` property when creating the [ShaderMaterial][1].
+
+### Uniforms
+
+Uniforms are used to pass *numerical resources* from the engine to the shader.
+
+Uniforms are declared using this simplified syntax:
+
+```wgsl
+uniform view_position: vec3f;
+uniform tints: array<vec3f, 4>;
+uniform weights: array<f32, 8>;
+```
+
+Internally, uniforms are automatically placed in a uniform buffers, and in the shader code are accessed using `uniform.` prefix:
+
+```wgsl
+var pos = uniform.view_position;
+var color = uniform.tints[2];
+
+// f32 and vec2<> types used in an array are due to alignment requirements wrapped
+// in an aligned structure, and the value is available as its `element` property.
+// struct WrappedF32 { @size(16) element: f32 }
+var weight = uniform.weights[3].element;
+```
+
+The engine automatically sets appropriate uniform values when rendering.
+
+:::note
+
+Currently, our uniform system supports only simple types, including `f32`, `i32`, `u32`, as well as vectors and matrices (e.g., `vec4f`, `mat4x4f`). Structs are not supported at this time, so all uniform values must be declared as individual variables of basic types.
+
+:::
+
+### Texture resources
+
+Texture resources are using simplified WGSL syntax, where specifying a @group and @binding index for each resource has to be omitted.
+
+#### Sampling Textures
+
+In WGSL, textures and samplers are treated as separate objects, unlike in GLSL, where those are combined.
+
+When you want to sample a texture (i.e. retrieve filtered texel values), you must provide a texture object *directly followed* by a sampler.
+
+```wgsl
+// 2d texture with a sampler declaration
+var diffuseMap: texture_2d<f32>;
+var diffuseMapSampler: sampler;
+
+// texture sampling
+var texel = textureSample(diffuseMap, diffuseMapSampler, coords);
+```
+
+#### Fetching textures
+
+If you only need to read raw texel data (i.e., without filtering, mipmapping, or addressing modes), you can use textureLoad instead of textureSample. This is called non-filtered access, or simply texel fetching.
```

Comment:
```suggestion
If you only need to read raw texel data (i.e., without filtering, mipmapping, or addressing modes), you can use `textureLoad` instead of `textureSample`. This is called non-filtered access, or simply texel fetching.
```